lib/senkyoshi/models/content_file.rb

```diff
@@ -26,10 +26,15 @@ def self.correct_linkname(canvas_file)
       canvas_file.file_path.split("/").last
     end
 
-    def canvas_conversion(canvas_file = nil)
-      path = canvas_file ? canvas_file.file_path : @linkname
+    def canvas_conversion(resources, canvas_file = nil)
+      path = if canvas_file
+               canvas_file.file_path
+             else
+               resource = resources.detect_xid(@name)
+               resource.path if resource
```

Comment:
http://www.rubydoc.info/gems/rubocop/0.43.0/RuboCop/Cop/Style/SafeNavigation cool, I didn't know you could do that.